### ISSUE TITLE
Unable to resolve module expo-modules-core in v2.5.1

### ISSUE THREAD

**Issue by jamesholcomb** [FaridSafi/react-native-gifted-chat#2516]
#### Issue Description

After upgrading from 2.4.0 to 2.5.1

I don't use expo.

```
error: Error: Unable to resolve module expo-modules-core from /Users/jamesholcomb/code/app/node_modules/expo-clipboard/build/Clipboard.js: expo-modules-core could not be found within the project or in these directories:
  node_modules
> 1 | import { EventEmitter, UnavailabilityError, Platform } from 'expo-modules-core';
    |                                                              ^
  2 | import { ClipboardPasteButton } from './ClipboardPasteButton';
  3 | import ExpoClipboard from './ExpoClipboard';
  4 | const emitter = new EventEmitter(ExpoClipboard);
```

#### Additional Information

System:
  OS: macOS 13.6.6
  CPU: (10) arm64 Apple M1 Max
  Memory: 179.72 MB / 32.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 20.14.0
    path: ~/.nvm/versions/node/v20.14.0/bin/node
  Yarn:
    version: 3.8.3
    path: ~/.nvm/versions/node/v20.14.0/bin/yarn
  npm:
    version: 10.7.0
    path: ~/.nvm/versions/node/v20.14.0/bin/npm
  Watchman:
    version: 2024.07.15.00
    path: /opt/homebrew/bin/watchman
Managers:
  CocoaPods:
    version: 1.15.2
    path: /opt/homebrew/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 23.2
      - iOS 17.2
      - macOS 14.2
      - tvOS 17.2
      - watchOS 10.2
  Android SDK: Not Found
IDEs:
  Android Studio: 2024.1 AI-241.18034.62.2411.12071903
  Xcode:
    version: 15.1/15C65
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 17.0.11
    path: /usr/bin/javac
  Ruby:
    version: 2.7.7
    path: /Users/jamesholcomb/.rbenv/shims/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.73.8
    wanted: 0.73.8
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: false
iOS:
  hermesEnabled: true
  newArchEnabled: false

**Comment by kesha-antonov** [comment]
Hi

Please try test with latest commit

`"react-native-gifted-chat": "FaridSafi/react-native-gifted-chat#206b88e6011482c3bed398ffaec83041e505a704"`

**Comment by kesha-antonov** [comment]
The problem:

we use Clipboard and it was removed from core https://reactnative.dev/docs/clipboard that 
So we have 2 options:
- use https://github.com/react-native-clipboard/clipboard
It's not compatible with expo so people with expo will not be able to use RNGC. We can write config to make it compatible

- use https://github.com/expo/expo/tree/main/packages/expo-clipboard
People who don't use expo would need to install expo and configure their project which may be not desirable.

Personally, I think we should go first way. But I need to figure out how to write expo module config

Related: https://github.com/kesha-antonov/react-native-background-downloader/issues/9

**Comment by dawpham3009** [comment]
Hi @kesha-antonov 
I have installed lasted version but in React Native CLI unsolved expo-modules althoughts it was removed. Please take for why reason?

**Comment by Yash-CBL** [comment]
I am using react native version@0.74.0 and upgraded the library to the latest version but still getting this error
expo-modules-core could not be found within the project or in these directories: node_modules

**Comment by kesha-antonov** [comment]
> The problem:
> 
> we use Clipboard and it was removed from core https://reactnative.dev/docs/clipboard that So we have 2 options:
> 
> * use https://github.com/react-native-clipboard/clipboard
>   It's not compatible with expo so people with expo will not be able to use RNGC. We can write config to make it compatible
> * use https://github.com/expo/expo/tree/main/packages/expo-clipboard
>   People who don't use expo would need to install expo and configure their project which may be not desirable.
> 
> Personally, I think we should go first way. But I need to figure out how to write expo module config
> 
> Related: [kesha-antonov/react-native-background-downloader#9](https://github.com/kesha-antonov/react-native-background-downloader/issues/9)

I think it will solve it. I'll try to fix it

**Comment by Nicholas1K** [comment]
using react native native CLI following the instructions listed above it gives me this error Unable to resolve module expo-modules-core can you tell me why and how to solve it?

**Comment by Nicholas1K** [comment]
I get this error using react native CLI: Unable to resolve module expo-modules-core from /Volumes/Nik_Tizzani/programmazione/app/HairPro_Organizer-main/node_modules/expo-clipboard/build/ExpoClipboardPasteButton.js: expo-modules-core could not be found within the project or in these directories: node_modules > 1 | import { requireNativeViewManager } from 'expo-modules-core';

**Comment by jamesholcomb** [comment]
Or option 3, remove the dependency altogether.  Copy to clipboard is a concern most likely already handled by apps.

**Comment by kesha-antonov** [comment]
> Or option 3, remove the dependency altogether.  Copy to clipboard is a concern most likely already handled by apps.

Yeah, good point also. It would be breaking change but easy to implement in apps. I think we should go this way, you're right

**Comment by arnekolja** [comment]
Well, at the moment it's a breaking change, too. So … 🙃
I think offering copy and paste per default is somewhat over-specific anyway, as most devs won't only implement that one, but others in context menus too. So it's a concern of theirs one way or another.

**Comment by kesha-antonov** [comment]
> Well, at the moment it's a breaking change, too. So … 🙃
> 
> I think offering copy and paste per default is somewhat over-specific anyway, as most devs won't only implement that one, but others in context menus too. So it's a concern of theirs one way or another.

Agreed.

**Comment by kesha-antonov** [comment]
I've made a fix. Will upload today

**Comment by vokecodes** [comment]
Cool. Waiting on the update

**Comment by kesha-antonov** [comment]
Fixed

- Please try version from master
- Follow installation guide https://github.com/FaridSafi/react-native-gifted-chat?tab=readme-ov-file#getting-started

**Comment by kesha-antonov** [comment]
Should be fixed in 2.6.0

**Comment by aashish-tanwar** [comment]
@kesha-antonov After upgrading to 2.6.0 from 2.4.0
I'm facing another issue

Do you mind helping with this?

<img width="1036" alt="Screenshot 2024-08-28 at 1 36 14 PM" src="https://github.com/user-attachments/assets/81e085c8-6efc-40e1-8457-40b7d47ec17f">
